nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/ui/flow-analysis-rules/flow-analysis-rule-table/flow-analysis-rule-table.component.ts

```diff
@@ -0,0 +1,261 @@
+
+    ngAfterViewInit(): void {
+        this.dataSource.sort = this.sort;
+    }
+
+    canRead(entity: FlowAnalysisRuleEntity): boolean {
+        return entity.permissions.canRead;
+    }
+
+    canWrite(entity: FlowAnalysisRuleEntity): boolean {
+        return entity.permissions.canWrite;
+    }
+
+    canOperate(entity: FlowAnalysisRuleEntity): boolean {
+        if (this.canWrite(entity)) {
+            return true;
+        }
+        return !!entity.operatePermissions?.canWrite;
+    }
+
+    hasComments(entity: FlowAnalysisRuleEntity): boolean {
+        return !this.nifiCommon.isBlank(entity.component.comments);
+    }
+
+    getCommentsTipData(entity: FlowAnalysisRuleEntity): TextTipInput {
+        return {
+            text: entity.component.comments
+        };
+    }
+
+    hasErrors(entity: FlowAnalysisRuleEntity): boolean {
+        return !this.nifiCommon.isEmpty(entity.component.validationErrors);
+    }
+
+    getValidationErrorsTipData(entity: FlowAnalysisRuleEntity): ValidationErrorsTipInput {
+        return {
+            isValidating: entity.status.validationStatus === 'VALIDATING',
+            validationErrors: entity.component.validationErrors
+        };
+    }
+
+    hasBulletins(entity: FlowAnalysisRuleEntity): boolean {
+        return !this.nifiCommon.isEmpty(entity.bulletins);
+    }
+
+    getBulletinsTipData(entity: FlowAnalysisRuleEntity): BulletinsTipInput {
+        return {
+            bulletins: entity.bulletins
+        };
+    }
+
+    getStateIcon(entity: FlowAnalysisRuleEntity): string {
+        if (entity.status.validationStatus === 'VALIDATING') {
+            return 'validating fa fa-spin fa-circle-o-notch';
+        } else if (entity.status.validationStatus === 'INVALID') {
+            return 'invalid fa fa-warning';
+        } else {
+            if (entity.status.runStatus === 'DISABLED') {
+                return 'disabled icon icon-enable-false';
+            } else if (entity.status.runStatus === 'DISABLING') {
+                return 'disabled icon icon-enable-false';
+            } else if (entity.status.runStatus === 'ENABLED') {
+                return 'enabled fa fa-flash';
+            } else if (entity.status.runStatus === 'ENABLING') {
+                return 'enabled fa fa-flash';
+            }
+        }
+        return '';
+    }
+
+    formatState(entity: FlowAnalysisRuleEntity): string {
+        if (entity.status.validationStatus === 'VALIDATING') {
+            return 'Validating';
+        } else if (entity.status.validationStatus === 'INVALID') {
+            return 'Invalid';
+        } else {
+            if (entity.status.runStatus === 'DISABLED') {
+                return 'Disabled';
+            } else if (entity.status.runStatus === 'DISABLING') {
+                return 'Disabling';
+            } else if (entity.status.runStatus === 'ENABLED') {
+                return 'Enabled';
+            } else if (entity.status.runStatus === 'ENABLING') {
```

Comment:
Compared to the existing NiFi UI, it doesn't appear that FlowAnalysisRules support `ENABLING` and `DISABLING` transitional states.